File: remove_element.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i = 0;
        // for each loop for array traversing and checking whether val is present in array or not
        for (const int num : nums)
            if (num != val)
                nums[i++] = num;
        
        return i;
    }
};

int main() {
    // Example usage
    Solution solution;

    // Create a vector of integers
    vector<int> nums = {3, 2, 2, 3, 4, 5, 3, 6};
    // Value to be removed
    int valToRemove = 3;

    // Call the removeElement function
    int newSize = solution.removeElement(nums, valToRemove);

    // Print the modified vector and its size
    cout << "Modified Vector: ";
    for (int i = 0; i < newSize; ++i) {
        cout << nums[i] << " ";
    }
    cout << "\nNew Size: " << newSize <<endl;

    return 0;
}

File: Remove_element.java
// Questions--->https://leetcode.com/problems/remove-element/

package leet_code;

public class Remove_element {
    public int removeElement(int[] nums, int val) {
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[count] = nums[i];
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Remove_element solution = new Remove_element();
        
        int[] nums = {3, 2, 2, 3};
        int val = 3;
        int newLength = solution.removeElement(nums, val);

        System.out.println("New length: " + newLength);
        System.out.print("Modified array: ");
        for (int i = 0; i < newLength; i++) {
            System.out.print(nums[i] + " ");
        }
        System.out.println();

        // Additional test case
        nums = new int[]{0, 1, 2, 2, 3, 0, 4, 2};
        val = 2;
        newLength = solution.removeElement(nums, val);

        System.out.println("New length: " + newLength);
        System.out.print("Modified array: ");
        for (int i = 0; i < newLength; i++) {
            System.out.print(nums[i] + " ");
        }
        System.out.println();
    }
}